### DISCUSSION TITLE
multiclass classification questions

### DISCUSSION THREAD

**Discussion by rxjx** [autogluon/autogluon#2338] [Q&A]
Two questions regarding multi-class tabular classification:

(1) I've labeled rows with no relevant classes as NaN. However, I think these rows are being ignored instead of being used as negative examples, based on this message I get during training:
`Warning: Ignoring 4675 (out of 18292) training examples for which the label value in column 'lbl' is missing`
Is there a way to force these to be seen as negative examples for all classes?

(2) Is there a way to get per-class metrics for test data? I can only see ways to get global metrics. Also, in a similar manner to the (1) above, it looks like I have to drop rows with NaN for labels in the test data. Ideally, one would want to keep that data as negative examples as well.

**Comment by Innixma** [comment]
1. We don't support negative examples. Please provide reasoning for why negative example support is important to improving the result, as well as an example of a popular tabular ML model that supports negative examples.
2. For now, you are best served calculating them yourself, we don't have a helper function to calculate per-class metrics.

**Reply by rxjx** [reply]
I don't have enough data about other tabular ML models so I can't speak to that. However, the motivating use case for me is to deal with FPs by adding similar data to the training set as a negative for all classes. My classes do not cover all possible images in the wild. So either I have a fake class similar to "none" or set label values to NaN. If there's a better way to deal with this please let me know.
